include/alpaka/rand/Philox/PhiloxSingle.hpp

```diff
@@ -0,0 +1,138 @@
+/* Copyright 2021 Jiri Vyskocil
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/rand/Philox/MultiplyAndSplit64to32.hpp>
+#include <alpaka/rand/Philox/PhiloxBaseTraits.hpp>
+
+#include <utility>
+
+namespace alpaka
+{
+    namespace rand
+    {
+        namespace engine
+        {
+            /** Philox state for single value engine
+             *
+             * @tparam TCounter Type of the Counter array
+             * @tparam TKey Type of the Key array
+             */
+            template<typename TCounter, typename TKey>
+            struct PhiloxStateSingle
+            {
+                using Counter = TCounter;
+                using Key = TKey;
+
+                Counter counter; ///< Counter array
+                Key key; ///< Key array
+                Counter result; ///< Intermediate result array
+                std::uint32_t position; ///< Pointer to the active intermediate result element
+                // TODO: Box-Muller states
```

Comment:
What needs to be done here? Please remove the TODO or explain in more detail what is missing.